doc/dev_doc/plugins.rst

```diff
@@ -216,6 +216,18 @@ Each hook available in INGInious is described here, starting with its name and p
 
     Adds a new helper to the instance of TemplateHelper. Should return a tuple (name,func) where name is the name that will
     be indicated when calling the TemplateHelper.call method, and func is the function that will be called.
+``feedback_text`` (``task``, ``submission``, ``text``)
```

Comment:
If I understand well, this feature is meant for allowing later modifications on (global included) feedback based on custom fields, right ?

It is interesting for global feedback but concerning subproblems feedback I remain convinced that it would be simpler and architecturally nicer to replace the old reload feedback function, while finishing the subproblems refactoring, so that they can be simply called with JSON, or a dict. That means that current feedback would be a JSON string (or be replaced by a dict) and that problems will either parser the json either simply display the feedback as it is working now.

Therefore highlighting a line in a code-problem can be performed by the code-problem feedback loading function. If a Python pre-processing is required, maybe we can add such a method the the DisplayableProblem class. This way feedback will only be treated in a unique process whatever the problem is. This would be more similar to what we've seen on Dodona : every problem has its own way displaying feedback.

Reply:
You mean having something like this in the return value of the container?
```
{ "id": "submission_id",
  #...
  "feedback": {
    "@global": {
      "text": "The global feedback text in RST"
      "custom_field_1": "a custom field"
      "custom_field_2": {"something": "another custom field with a dict"}
    },
    "subproblem_1": {
      "result": "ok",
      "text": "the feedback for sp1",
      "custom_field3": "a custom field only for this subproblem"
    },
    #...
  }
}
```